fc-community-contracts/test/FactchainNFT.t.sol

```diff
@@ -7,31 +7,48 @@ import {FactchainProxy} from "../src/FactchainProxy.sol";
 
 contract FactchainNFTTest is Test {
     FactchainNFT collection;
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
 
     address public owner = address(1);
-    address public recipient = address(2);
-    address public factChainSFT = address(3);
-    address[] public raters = [address(4), address(5)];
+    address public admin = address(2);
+    address public recipient = address(3);
+    address public factChainSFT = address(4);
+    address[] public raters = [address(5), address(6)];
     string public postUrl = "https://twitter.com/rektorship/status/1750724519705153835";
 
     function setUp() public {
         FactchainNFT implementation = new FactchainNFT();
         FactchainProxy proxy =
         new FactchainProxy(address(implementation), abi.encodeCall(collection.initialize, (owner, factChainSFT, "https://gateway.pinata.cloud/ipfs/")));
         collection = FactchainNFT(payable(address(proxy)));
+
+        hoax(owner);
+        collection.grantRole(MINTER_ROLE, admin);
     }
 
-    function testGetTokenURI() public {
+    function test_mint() public {
+        vm.prank(admin);
+        collection.mint(recipient, postUrl, raters, "QmdByzCXRCbPZ2DYq3wqURjShNEGZ2qDNW4w9mcwA9oWLv");
+    }
+
+    function test_mint_RevertIf_notAdmin() public {
+        vm.expectRevert();
         vm.prank(owner);
         collection.mint(recipient, postUrl, raters, "QmdByzCXRCbPZ2DYq3wqURjShNEGZ2qDNW4w9mcwA9oWLv");
+    }
+
+    function testGetTokenURI() public {
```

Comment:
nitpick, but we usually write `test_getTokenURl`

Reply:
yep np